### PR TITLE
chore!: update packages

### PR DESCRIPTION
**Pull request by saul-jb**
This PR updates all the libp2p related packages to make it work with the newer versions of libp2p.

* Fixes the silent bootstrap replicator fail with the new Libp2p.

The breaking changes is the libp2p streams interface that has changed - as far as I can tell this only breaks the bootstrap replicator.

### CODE REVIEW

**Review by tabcat — COMMENTED** [review]
Looks good :+1: just the one comment and we can get this merged :100: 

Also, wondering if it was necessary to bump zzzync with the libp2p bump

File: test/utils/libp2p-services.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
-
-import { gossipsub, type GossipSub } from '@chainsafe/libp2p-gossipsub'
+import { gossipsub } from '@chainsafe/libp2p-gossipsub'
+import type { PubSub } from '@libp2p/interface-pubsub'
```
[the remark this answers is not in the document]
Comment (saul-jb) [reply test/utils/libp2p-services.ts]: Yes, I've fixed this along with all the others.

**Comment by saul-jb** [comment]
> Also, wondering if it was necessary to bump zzzync with the libp2p bump

Yeah, the old zzzync uses the older interfaces which were giving errors.

**Comment by tabcat** [comment]
plan to refactor tests and test utils in the future so everything is less flaky, already started https://github.com/hldb/welo/pull/103 but thinking about refactoring code first and closing that PR

**Pull request merged**